### ISSUE TITLE
need to run the molecule tests periodically

### ISSUE THREAD

**Issue by jmazzitelli** [kiali/kiali#2591]
We have a good suite of molecule tests. These are currently run manually only. We should get these to run automatically through some CI job.

**Comment by jmazzitelli** [comment]
One thing to note - if running on an OpenShift cluster and the tests will be done with dev images (`MOLECULE_USE_DEV_IMAGES=true`), this requires the dev images pushed to the 'kiali' namespace to be allowed to be pulled into the `anothernamespace` namespace as part of the test `default-namespace-test`. Otherwise, the kiali pod will not be authorized to pull the kiali dev image and this error results: 
```
Failed to pull image "image-registry.openshift-image-registry.svc:5000/kiali/kiali:dev": rpc error: code = Unknown desc = Error reading manifest dev in image-registry.openshift-image-registry.svc:5000/kiali/kiali: unauthorized: authentication required
```
In short, if a test installs Kiali in any namespace other than istio-system, then you have to give authorization to that other namespace because the current Makefile only prepares authorization to the namespace you give it in the cluster-push target. So something like this is needed:
```
oc policy add-role-to-user system:image-puller system:serviceaccount:anothernamespace:kiali-service-account --namespace=kiali 
```

**Comment by jmazzitelli** [comment]
The PR https://github.com/kiali/kiali/pull/2943 provides a script that can run any set of molecule tests that you want. It performs the necessary preparations for tests that need it (like that "default-namespace-test" I mentioned in the earlier comment).

**Comment by jmazzitelli** [comment]
If you install OpenShift on a bare metal machine using https://github.com/kxr/ocp4_setup_upi_kvm (taking most of the defaults) then you can run the molecule tests by installing a script like this in the machine's root crontab (or, if not root, you need to give the user read/execute access to everything under /root/ocp4_setup_ocp4):

```
#!/bin/bash

# abort on any error
set -e

ID="/root/ocp4_setup_ocp4/install_dir"
OC="/root/ocp4_setup_ocp4/oc"
SRC="/tmp/KIALI-TESTS"
DORP="podman"

echo "Create a clean github repo location"
rm -rf /tmp/KIALI-TESTS
mkdir -p $SRC

echo "Make sure everything exists"
test -d $ID || (echo "$ID is missing"; exit 1)
test -x $OC || (echo "$OC is missing"; exit 1)
test -d $SRC || (echo "$SRC is missing"; exit 1)
which $DORP > /dev/null || (echo "$DORP is not in the PATH"; exit 1)

echo "Clone github repos to make sure we have the latest tests and scripts"
cd $SRC
git clone git@github.com:kiali/kiali
git clone git@github.com:kiali/kiali-operator kiali/operator
cd kiali

echo "Log into the cluster"
$OC login -u kubeadmin -p $(cat $ID/auth/kubeadmin-password) https://api.ocp4.local:6443

echo "Run the tests"
hack/run-molecule-tests.sh --client-exe "$OC"
```

**Comment by jmazzitelli** [comment]
To run molecule tests on minikube, PR https://github.com/kiali/kiali/pull/2945/files provides this script that can be run as a cronjob: `hack/ci-minikube-molecule-tests.sh`
